### ISSUE TITLE
getNextNonce should return `BigInteger.ONE` if account doesn't exist

### ISSUE THREAD

**Issue by marc0olo** [kryptokrauts/contraect-maven-plugin#66]
I tried to perform `dry-run` calls and those always failed because `getNextNonce()` doesn't handle `Account not found` properly

![image](https://user-images.githubusercontent.com/8124114/125615901-ac67165e-df19-41f4-8c31-96491c0f1c45.png)

we should return `BigInteger.ONE` if the account does not exist yet which should allow us to perform the dry-run call

**Comment by marc0olo** [comment]
would be good to have this fixed quickly because I want to add a new example to the showcase repo which requires this as it runs against mainnet (TokenMigration contract)

**Comment by marc0olo** [comment]
@mitch-lbw I think we can close this. just tried the workaround and I am getting an error for the dry-run call:

- `Internal error: insufficient_funds`

so it probably doesn't matter. maybe it's possible to provide an option to the dry-run endpoint to ignore insufficient funds

**Comment by marc0olo** [comment]
seems like it is possible to provide an amount for dry-run calls, see discussion here https://github.com/aeternity/aeternity/issues/3677

**Comment by marc0olo** [comment]
so the bug here is still valid as we need to cover `Account not found` anyway.

@mitch-lbw we need to think about how and where we want to provide this. it probably makes sense to additionally introduce a new config-param for the `AeternityServiceConfiguration` in the SDK where a user can optionally define an amount to be used for the account that performs a dry-run call.

**Comment by marc0olo** [comment]
amount should also be quite easy to add:

```java
  private DryRunTransactionResult dryRunCall(ContractCallTransactionModel contractCallModel,
      String functionName) {
    DryRunTransactionResults dryRunResults = aeternityService.transactions.blockingDryRunTransactions(DryRunRequest.builder().build().transactionInputItem(contractCallModel).account(DryRunAccountModel.builder().amount(new BigInteger("100000000000000000000000000000000000").publicKey(this.config.getKeyPair().getAddress()).build()));
    if(dryRunResults.getResults() != null && dryRunResults.getResults().size()>0) {
      return dryRunResults.getResults().get(0);
    }
    Object dryRunErrorMessage = dryRunResults.getResults().get(0).getContractCallObject().getReturnValue();
    try{dryRunErrorMessage=this.aeternityService.compiler.blockingDecodeCallResult(aesSourcecode,functionName,dryRunResults.getResults().get(0).getContractCallObject().getReturnType(),dryRunResults.getResults().get(0).getContractCallObject().getReturnValue(),aesIncludes).getResult();} catch(Exception e){}throw new RuntimeException(String.format("\n"
            + "Dry run call of function %s failed: %s\n"
            + "Causing Exception: %s%nException Details: %s\n"
            + "Exception Hint   : %s",functionName,dryRunErrorMessage,dryRunResults.getAeAPIErrorMessage(),dryRunResults.getRootErrorMessage(),"Please validate your input data"));
  }
  ```

in the snippet above (generated by the plugin) I added `.amount(new BigInteger("100000000000000000000000000000000000")` to the builder of the `DryRunAccountModel`

## note
- so just simply adding an amount to the `DryRunAccount` probably does work if the `amount` is enough to cover the potential fees, e.g. `BigInteger.TEN` (which I tested first) obviously wasn't enough

**Comment by marc0olo** [comment]
the javascript sdk e.g. always uses the zero address with a pre-defined amount for every dry-run call.

see:
- https://github.com/aeternity/aepp-sdk-js/blob/80d0879e230a2df1fa4c9bfc8aa681de4f95940c/src/tx/builder/schema.js#L28
- https://mainnet.aeternity.io/v2/accounts/ak_11111111111111111111111111111111273Yts (nonce is always `1`)
- https://github.com/aeternity/aepp-sdk-js/blob/80d0879e230a2df1fa4c9bfc8aa681de4f95940c/src/ae/contract.js#L123

doing it that way would allow us to skip the requirement of having a configured keypair and we could use this for every static call. the `next nonce` would always be `2` for that account

**Comment by mitch-lbw** [comment]
Will be fixed with https://github.com/kryptokrauts/aepp-sdk-java/issues/173

**Comment by marc0olo** [comment]
@mitch-lbw can/should we close this then?

**Comment by mitch-lbw** [comment]
Yes we can close this issue

**Comment by mitch-lbw** [comment]
fixed with kryptokrauts/aepp-sdk-java#173
